pkg/publish/tarball.go

```diff
@@ -51,6 +51,12 @@ func (t *tar) Publish(_ context.Context, br build.Result, s string) (name.Refere
 	// https://github.com/google/go-containerregistry/issues/212
 	s = strings.ToLower(s)
 
+	// safeguarding against the case where the user has specified an empty tag with publish.WithTags([]string{""})
+	if len(t.tags) == 0 {
+		log.Printf("No tags specified, using default tag: %s", latestTag)
+		t.tags = []string{latestTag}
+	}
```

Comment:
The comment here and elsewhere doesn't match the code. `[]string{""}` is different from `[]string{}`; the first has `len(tags) == 1`, the second has `len == 0` and matches the code.

I'm not sure what happens if passed `WithTags([]string{""})` or for that matter `WithTags([]string{"", "", "", "latest", ""})`. It's probably worth finding out and fixing any confusion that might cause. Doesn't need to block this PR though.

Reply:
> The comment here and elsewhere doesn't match the code. []string{""} is different from []string{}; the first has len(tags) == 1, the second has len == 0 and matches the code.

My bad 🤦 I've fixed it.